README.md

```diff
@@ -39,6 +72,70 @@ The IPFS daemon attempts to do this on its own, but only up to 8192 FDs by defau
 This is controllable via the `IPFS_FD_MAX` environment variable.
 Setting `IPFS_FD_MAX="100000"` should be sufficient.
 
+## Logs
+
+This plugin uses the usual IPFS logging API.
+To see logs produced by this plugin, either set an appropriate global log level (the default is `ERROR`):
+```
+IPFS_LOGGING="info" ipfs daemon
+```
+
+Or, after having started the daemon, configure just this component to emit logs at a certain level or above:
+```
+ipfs daemon&
+ipfs log level metric-export info
+ipfs log tail # or something else?
+```
+
+## The TCP Server
+
+This plugin comes with a TCP server that pushes Bitswap messages and other things to clients.
+The protocol consists of 4-byte, big endian, framed, JSON-encoded messages.
+Thus, on the wire, each message looks like this:
+```
+<size of following message in bytes, 4 bytes big endian><JSON-encoded message>
+```
+
+A connection starts with a handshake, during which both sides send a version message of this form:
+```go
+// A version message, exchanged between client and server once, immediately
+// after the connection is established.
+type versionMessage struct {
+	Version int `json:"version"`
+}
+```
+Encoded in the usual manner.
+Both sides verify that the version matches.
+If there is a mismatch, the connection is closed.
+
+A client is, by default, subscribed to events emitted by this plugin.
+Messages sent by this plugin are of this format:
+```go
+// The type sent out via TCP.
+type outgoingTCPMessage struct {
+	// If Event is not nil, this message is a pushed event.
+	Event *event `json:"event"`
+}
+
+// The type sent to via TCP for pushed events.
+type event struct {
+	// The timestamp at which the event was recorded.
+	// This defines an ordering for events.
+	Timestamp time.Time `json:"timestamp"`
+
+	// Peer is a base58-encoded string representation of the peer ID.
+	Peer string `json:"peer"`
+
+	// BitswapMessage is not nil if this event is a bitswap message.
+	BitswapMessage *BitswapMessage `json:"bitswap_message,omitempty"`
+
+	// ConnectionEvent is not nil if this event is a connection event.
+	ConnectionEvent *ConnectionEvent `json:"connection_event,omitempty"`
```

Comment:
So we ditch the export of currently connected peers and solve this via ConnectionEvents?

Reply:
We can get those via the usual IPFS API, no? Or is that significantly worse?